clients/fides-js/src/lib/preferences.ts

```diff
@@ -89,29 +78,31 @@ export const updateConsentPreferences = async ({
   };
   patchUserPreferenceToFidesServer(privacyPreferenceCreate, fidesApiUrl, debug);
 
-  // 2. Update the window.Fides.consent object
+  // 2. Update the cookie object based on new preferences
+  const updatedCookie = await updateCookie(cookie);
+  Object.assign(cookie, updatedCookie);
```

Comment:
I learned something weird— assigning to the function param with something like

```
param = updatedParam;
```
doesn't work! (`param` in the parent scope is unchanged). However, doing an `Object.assign` does work, and so does assigning to individual attributes such as

```
param.consent = updatedParam.consent
```